### PR TITLE
IT IS DONE

### PR DESCRIPTION
**Pull request by javierbustillo**
Ugly af but you can upload a file and then it is shown

### CODE REVIEW

File: src/js/components/Chat.jsx [hunk 1]
```diff
@@ -28,8 +30,11 @@ class ConnectedChat extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            message: ''
+            message: '',
+            picture: []
         }
+        this.state = { pictures: [] };
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx]: Overrides above 
```python
this.state
```

File: src/js/components/Chat.jsx [hunk 2]
```diff
@@ -28,8 +30,11 @@ class ConnectedChat extends Component {
...
+        this.onDrop = this.onDrop.bind(this);
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx]: No need to bind

File: src/js/components/Chat.jsx [hunk 3]
```diff
@@ -51,17 +56,29 @@ class ConnectedChat extends Component {
     postMessage = (event) => {
         event.preventDefault();
         const message = this.state.message;
-        this.props.postMessage(message);
+        this.props.postMessage({message: message, picture: this.state.picture[0]});
         this.setState({message: ''});
+
     };
 
+    onDrop(picture) {
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx:62]:
```python
OnDrop = (picture) => {
```

File: src/js/components/Chat.jsx [hunk 4]
```diff
@@ -51,17 +56,29 @@ class ConnectedChat extends Component {
...
+        this.setState({
+            picture: this.state.pictures.concat(picture),
+        });
+    }
+
     renderMessage = m => {
-        const {mid, created_on, message, uid, likes, dislikes, image} = m;
+        const {mid, created_on, message, uid, likes, dislikes, image, img} = m;
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx]: Why add “img” when there is already an “image” variable

File: src/js/components/Chat.jsx [hunk 5]
```diff
@@ -51,17 +56,29 @@ class ConnectedChat extends Component {
...
         const message = this.state.message;
-        this.props.postMessage(message);
+        this.props.postMessage({message: message, picture: this.state.picture[0]});
         this.setState({message: ''});
+
     };
 
+    onDrop(picture) {
+        this.setState({
+            picture: this.state.pictures.concat(picture),
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx:64]: Should create an action type, action, and reducer for this

File: src/js/components/Chat.jsx [hunk 6]
```diff
@@ -75,7 +92,7 @@ class ConnectedChat extends Component {
                                                      alt="Card image cap"/>
                                     : <br/>}
                                 <CardBody>
-                                    <CardText>{message}</CardText>
+                                    <CardText>{message}</CardText>s
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx]: Remove additional “s” at the end of this line

File: src/js/components/Chat.jsx [hunk 7]
```diff
@@ -96,6 +113,8 @@ class ConnectedChat extends Component {
                                 {/*TODO: Detect if message has image to display in chat.*/}
                                 {/*<CardImg top width="100%" src="https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180" alt="Card image cap" />*/}
                                 <CardBody>
+                                    {recentImage ? <CardImg top height="100%" width="100%" src={imgSource}
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx:113]: This would make all messages have images

File: yarn.lock [hunk 8]
```diff
@@ -9740,7 +9753,7 @@ react-transition-group@^2.3.1:
     prop-types "^15.6.2"
     react-lifecycles-compat "^3.0.4"
 
-react@^16.8.1:
+react@^16.2.0, react@^16.8.1:
```
Comment (ChristianPerez34) [inline yarn.lock:9756]: What happened here?

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]
Some adjustments needed

File: src/js/components/Chat.jsx (hunk 7 above)
Comment (javierbustillo) [reply src/js/components/Chat.jsx:113]: Only if the message has a recentImage. Hence el if

File: yarn.lock (hunk 8 above)
Comment (javierbustillo) [reply yarn.lock:9756]: idk man tu me dijiste que le haga commit

File: src/js/components/Chat.jsx [hunk 9]
```diff
@@ -68,16 +66,16 @@ class ConnectedChat extends Component {
     }
 
     renderMessage = m => {
-        const {mid, created_on, message, uid, likes, dislikes, image, img} = m;
+        const {mid, created_on, message, uid, likes, dislikes, image, uploaded_image} = m;
```
Comment (ChristianPerez34) [inline src/js/components/Chat.jsx:69]: You cannot reuse 
```python 
image 
``` 
?

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: src/js/components/Chat.jsx (hunk 9 above)
Comment (javierbustillo) [reply src/js/components/Chat.jsx:69]: Screen Shot 2019-04-05 at 9 07 08 PM

Comment (javierbustillo) [reply src/js/components/Chat.jsx:69]: If I use image para los dos, se rompen los incoming messages. No tiene forma de saber si es una foto que viene del browser o es una foto que viene de un URL

Comment (ChristianPerez34) [reply src/js/components/Chat.jsx:69]:
```javascript
image = typeof image.name == 'string' ? URL.createObjectURL(image): image;
```
or
```javascript
image = image instanceof File ? URL.createObjectURL(image): image;
```

File: src/js/components/Chat.jsx (hunk 6 above)
Comment (ChristianPerez34) [reply src/js/components/Chat.jsx]:
```html
<CardText>{message}</CardText>
```

**Review by ChristianPerez34 — APPROVED** [review]

**Pull request merged**